src/journal-remote/journal-upload.c

```diff
@@ -448,6 +491,55 @@ static void destroy_uploader(Uploader *u) {
         sd_event_unref(u->event);
 }
 
+#if LIBCURL_VERSION_NUM >= 0x075300
+static int update_content_encoding(Uploader *u, char *acceptEncoding) {
+        bool found = false;
+        Compression algorithm;
+        CURLcode code;
+
+        assert(u);
+
+        while (acceptEncoding && !found) {
+                _cleanup_free_ char *encodingValue = NULL;
+                char *algorithmStr = NULL;
```

Comment:
Please also enable the cleanup attribute for `algorithmStr`.

Please rename variables to lower case with under bar.